### DISCUSSION TITLE
question about *.web extensions

### DISCUSSION THREAD

**Discussion by sgrund14** [chen-rn/CUA#42] [General]
hello! great repo thank you for making :)

have a small question about the files [here](https://github.com/chen-rn/CUA/tree/main/packages/app/utils)

where does the `*.web.ts` come into play? I noticed you have `clerk`/`clerk.web`, `auth`/`auth.web`, `trpc`/`trpc.web`, but the only place you actually use this distinction explicitly is for `trpc.web` [here](https://github.com/chen-rn/CUA/blob/main/apps/nextjs/pages/_app.tsx). are `clerk.web` and `auth.web` ever necessary? or does clerks expo client work cross platform?

**Comment by chen-rn** [comment]
Good question! When you make an import say... to "app/utils/clerk", if you're on web, react-native-web will automatically figure out if there's a .web file and use that instead. So while we don't see a direct import, the web app _is_ using the web auth! ie. Expo client for clerk does not work cross platform!

And the reason why we have a direct import to trpc.web in the example you pointed out is because we are on the nextjs side of things. Without it, I'd imagine it _should_ still work?

**Reply by sgrund14** [reply]
ah, I see! thank you kindly 🙏 💯
